src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperParseTreeRewriter.cs

```diff
@@ -324,12 +324,25 @@ private IEnumerable<string> GetAttributeNames(Block tagBlock)
                     childSpan = child as Span;
                 }
 
-                var attributeName = childSpan
-                    .Content
-                    .Split(separator: new[] { '=' }, count: 2)[0]
-                    .TrimStart();
+                var start = 0;
+                var end = 0;
+                for (; start < childSpan.Content.Length; start++)
```

Comment:
Does accessing `childSpan.Content` cause revaluation of a buffer? Maybe read it into a variable.